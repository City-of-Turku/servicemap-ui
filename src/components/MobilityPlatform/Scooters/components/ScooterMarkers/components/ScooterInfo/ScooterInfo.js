import { Link, Typography } from '@material-ui/core';
import PropTypes from 'prop-types';
import React from 'react';

const ScooterInfo = ({ classes, intl, item }) => {
  const titleTypo = messageId => (
    <div>
      <Typography variant="subtitle1">
        {intl.formatMessage({
          id: messageId,
        })}
      </Typography>
    </div>
  );

  const singleValTypo = (messageId, value) => (
    <div className={classes.marginTop}>
      <Typography variant="body2">
        {intl.formatMessage({
          id: messageId,
        })}
        :
        {' '}
        {value}
      </Typography>
    </div>
  );

  const renderStatus = (scooterStatus) => {
    if (!scooterStatus) {
      return (
        <div className={classes.marginTop}>
          <Typography variant="body2">
            {intl.formatMessage({ id: 'mobilityPlatform.content.scooter.notReserved' })}
          </Typography>
        </div>
      );
    } return null;
  };

  const renderLink = (linkUrl, text) => (
    <div className={classes.paragraph}>
      <Link target="_blank" href={linkUrl}>
        <Typography className={classes.link} variant="body2">
          {text}
        </Typography>
      </Link>
    </div>
  );

  const formatRange = (range) => {
    const rangeKm = (range / 1000).toFixed(2);
    return `${rangeKm} km`;
  };

  return (
    <div className={classes.container}>
      <div className={classes.headerContainer}>
        {titleTypo('mobilityPlatform.content.scooter.title')}
      </div>
      <div className={classes.textContainer}>
        {singleValTypo('mobilityPlatform.content.general.provider', 'Ryde')}
        {renderStatus(item.is_reserved)}
        {singleValTypo('mobilityPlatform.content.scooter.range', formatRange(item.current_range_meters))}
        <div className={classes.paragraph}>
          <Typography variant="body2" className={classes.bold}>
            {intl.formatMessage({ id: 'mobilityPlatform.content.general.rentalUris' })}
            :
          </Typography>
        </div>
        {renderLink(item.rental_uris.android, 'Android')}
        {renderLink(item.rental_uris.ios, 'iOS')}
      </div>
    </div>
  );
};

ScooterInfo.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  intl: PropTypes.objectOf(PropTypes.any).isRequired,
  item: PropTypes.objectOf(PropTypes.any),
};

ScooterInfo.defaultProps = {
  item: {},
};

export default ScooterInfo;
