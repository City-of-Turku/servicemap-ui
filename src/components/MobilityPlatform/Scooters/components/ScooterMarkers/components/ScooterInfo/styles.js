export default theme => ({
  container: {
    margin: theme.spacing(1),
  },
  headerContainer: {
    width: '85%',
    borderBottom: '1px solid #000',
    paddingBottom: theme.spacing(0.5),
  },
  textContainer: {
    marginTop: theme.spacing(0.5),
  },
  paragraph: {
    marginTop: theme.spacing(0.4),
  },
  marginTop: {
    marginTop: theme.spacing(0.5),
  },
  bold: {
    fontWeight: 'bold',
  },
  link: {
    color: theme.palette.link.main,
    textDecoration: 'underline',
  },
});
