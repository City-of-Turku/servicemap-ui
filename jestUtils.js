import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });