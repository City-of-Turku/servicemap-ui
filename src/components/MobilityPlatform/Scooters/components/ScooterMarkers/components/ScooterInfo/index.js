import { withStyles } from '@material-ui/core';
import { injectIntl } from 'react-intl';
import ScooterInfo from './ScooterInfo';
import styles from './styles';

export default withStyles(styles)(injectIntl(ScooterInfo));
