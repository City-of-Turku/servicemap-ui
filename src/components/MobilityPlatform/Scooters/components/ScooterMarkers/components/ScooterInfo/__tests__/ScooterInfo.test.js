// Link.react.test.js
import React from 'react';
import { getRenderWithProviders } from '../../../../../../../../../jestUtils';
import finnishTranslations from '../../../../../../../../i18n/fi';
import ScooterInfo from '../index';

const mockProps = {
  item: {
    bike_id: 'abc123test',
    is_reserved: false,
    current_range_meters: 20200,
    rental_uris: {
      android: 'https://www.testilinkki.fi',
      ios: 'https://www.testilinkki.fi',
    },
  },
};

const renderWithProviders = getRenderWithProviders({});

describe('<ScooterInfo />', () => {
  it('should work', () => {
    const { container } = renderWithProviders(<ScooterInfo {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<ScooterInfo {...mockProps} />);

    const p = container.querySelectorAll('p');
    const h6 = container.querySelector('h6');
    expect(h6.textContent).toContain(finnishTranslations['mobilityPlatform.content.scooter.title']);
    expect(p[0].textContent).toContain(`${finnishTranslations['mobilityPlatform.content.general.provider']}: Ryde`);
    expect(p[1].textContent).toContain(finnishTranslations['mobilityPlatform.content.scooter.notReserved']);
    expect(p[2].textContent).toContain(`${finnishTranslations['mobilityPlatform.content.scooter.range']}: 20.20 km`);
    expect(p[3].textContent).toContain(finnishTranslations['mobilityPlatform.content.general.rentalUris']);
  });

  it('does render links correctly', () => {
    const { container } = renderWithProviders(<ScooterInfo {...mockProps} />);

    const link = container.querySelectorAll('a');
    expect(link[0].textContent).toEqual('Android');
    expect(link[1].textContent).toEqual('iOS');
  });
});
