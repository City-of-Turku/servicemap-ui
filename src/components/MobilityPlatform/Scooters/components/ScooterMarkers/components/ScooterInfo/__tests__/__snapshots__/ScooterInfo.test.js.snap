// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScooterInfo /> should work 1`] = `
<div>
  <div
    class="injectIntl(ScooterInfo)-container-1"
  >
    <div
      class="injectIntl(ScooterInfo)-headerContainer-2"
    >
      <div>
        <h3
          class="MuiTypography-root MuiTypography-subtitle1 css-dlbwt1-MuiTypography-root"
        >
          Sähköpotkulauta
        </h3>
      </div>
    </div>
    <div
      class="injectIntl(ScooterInfo)-textContainer-3"
    >
      <div
        class="injectIntl(ScooterInfo)-marginTop-5"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
        >
          Palveluntarjoaja: Ryde
        </p>
      </div>
      <div
        class="injectIntl(ScooterInfo)-marginTop-5"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
        >
          Vuokrattavissa oleva
        </p>
      </div>
      <div
        class="injectIntl(ScooterInfo)-marginTop-5"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
        >
          Jäljellä oleva kantama: 20.20 km
        </p>
      </div>
      <div
        class="injectIntl(ScooterInfo)-paragraph-4"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 injectIntl(ScooterInfo)-bold-6 css-1ak20dk-MuiTypography-root"
        >
          Varauslinkit
          :
        </p>
      </div>
      <div
        class="injectIntl(ScooterInfo)-paragraph-4"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-16pak0r-MuiTypography-root-MuiLink-root"
          href="https://www.testilinkki.fi"
          role="link"
          target="_blank"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 injectIntl(ScooterInfo)-link-7 css-1ak20dk-MuiTypography-root"
          >
            Android
          </p>
        </a>
      </div>
      <div
        class="injectIntl(ScooterInfo)-paragraph-4"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-16pak0r-MuiTypography-root-MuiLink-root"
          href="https://www.testilinkki.fi"
          role="link"
          target="_blank"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 injectIntl(ScooterInfo)-link-7 css-1ak20dk-MuiTypography-root"
          >
            iOS
          </p>
        </a>
      </div>
    </div>
  </div>
</div>
`;
